Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Home.kt

```diff
@@ -108,21 +109,57 @@ enum class HomeSections(
     PROFILE(R.string.home_profile, Icons.Outlined.AccountCircle, "home/profile")
 }
 
+/**
+ * Remembers and creates an instance of [BottomBarStateHolder]
+ */
+@Composable
+fun rememberBottomBarStateHolder(navController: NavController, tabs: Array<HomeSections>) =
+    remember(navController, tabs) {
+        BottomBarStateHolder(navController, tabs)
+    }
+
+/**
+ * Responsible for holding state related to [JetsnackBottomBar] and containing UI-related logic.
+ */
+@Stable
+class BottomBarStateHolder(
+    private val navController: NavController,
+    val tabs: Array<HomeSections>
+) {
+    private val sections = HomeSections.values()
+    private val currentRoute: String?
+        get() = navController.currentDestination?.route
+
+    val routes = sections.map { it.route }
+
+    // Reading this attribute will cause recompositions when the route becomes valid or invalid
+    val isValidRoute: Boolean
+        @Composable get() =
+            navController.currentBackStackEntryAsState().value?.destination?.route in routes
+
+    val currentSection: HomeSections
+        get() = sections.first { it.route == currentRoute }
```

Comment:
Can this getter fail if the user is on a different route?

Reply:
If this is called when shouldn't be, it'll throw an exception. I think it's fine